rocketpy/AeroSurface.py

```diff
@@ -368,11 +313,94 @@ def evaluate_geometrical_parameters(self):
         -------
         None
         """
+
+        # If base radius is not given, the ratio between base radius and
+        # rocket radius is assumed as 1, meaning that the nose cone has the
+        # same radius as the rocket
         if self.base_radius is None or self.rocket_radius is None:
             self.radius_ratio = 1
+        # If base radius is given, the ratio between base radius and rocket
+        # radius is calculated
         else:
             self.radius_ratio = self.base_radius / self.rocket_radius
 
+        self.fineness_ratio = self.length / (2 * self.base_radius)
+        return None
+
+    def evaluate_nose_shape(self):
+        """Calculates and saves nose cone's shape. The shape is saved as two
+        vectors, one for the x coordinates and one for the y coordinates.
+
+        Returns
+        -------
+        None
+        """
+        # Constants
+        n = 127  # Points on the final curve.
+        p = 3  # Density modifier. Greater n makes more points closer to 0. n=1 -> points equally spaced.
+
+        # Calculate a function to find the tangential intersection point between the circle and nosecone curve.
+        def find_x_intercept(x):
+            return x + self.y_nosecone(x) * self.y_nosecone.differentiate(x)
+
+        # Calculate a function to find the radius of the nosecone curve
+        def find_radius(x):
+            return (self.y_nosecone(x) ** 2 + (x - find_x_intercept(x)) ** 2) ** 0.5
+
+        # Check bluffness circle and choose whether to use it or not
+        if self.bluffness is None or self.bluffness == 0:
```

Comment:
Keep in mind that matching the `self.bluffness == 0`  condition is really hard. 
Easier solution is to set a tolerance (let's say, 1e-8) and check if the value is inside the interval.

numpy.isclose

Reply:
This 0 is just to check when the use inputs the blufness as 0 (which is the same as None). Having a value of 1e-8 to the bluffness does not really make sense but should still work

Reply:
ok